src/TwigComponent/doc/index.rst

```diff
@@ -1058,6 +1058,141 @@ Exclude specific attributes:
       My Component!
     </div>
 
+Component with Complex Variants (CVA)
+-------------------------------------
+
+CVA is a concept from the JS words (https://cva.style/docs/getting-started/variants).
+It's a concepts use by the famous shadcn/ui library (https://ui.shadcn.com).
+This concept allow you to display a component with different variants (color, size, etc.),
+to create highly reusable and customizable components.
+You can use the cva function to define variants for your component.
+The first argument of the cva function is the base classes, the classes present in all variants. Then the second
+argument is an array of variants.
+
+.. code-block:: html+twig
+
+    {# templates/components/Alert.html.twig #}
+    {% props color = 'blue', size = 'md' %}
+
+     {% set alert = cva('alert', {
+        color: {
+            blue: 'bg-blue',
+            red: 'bg-red',
+            green: 'bg-green',
+        },
+        size: {
+            sm: 'text-sm',
+            md: 'text-md',
+            lg: 'text-lg',
+        }
+    }) %}
+
+    <div class="{{ alert.apply({color, size}) }} {{ attributes.render('class') }}">
+         {% block content %}{% endblock %}
+    </div>
+
+
+    {# index.html.twig #}
+
+    <twig:Alert color="red" size="lg">
+        <div>My content</div>
+    </twig:Alert>
+    // class="alert bg-red text-lg"
+
+    <twig:Alert color="green" size="sm">
+        <div>My content</div>
+    </twig:Alert>
+    // class="alert bg-green text-sm"
+
+    <twig:Alert class="flex items-center justify-center">
+        <div>My content</div>
+    </twig:Alert>
+    // class="alert bg-blue text-md flex items-center justify-center"
+
+CVA and Tailwind CSS
+~~~~~~~~~~~~~~~~~~~~
+
+CVA work perfectly with tailwindcss. The only drawback is you can have class conflicts,
+to have a better control you can use this following bundle (
+https://github.com/tales-from-a-dev/twig-tailwind-extra
+) in addition to the cva function:
+
+.. code-block:: terminal
+
+    $ composer require tales-from-a-dev/twig-tailwind-extra
+
+.. code-block:: html+twig
+
+    {# templates/components/Alert.html.twig #}
+    {% props color = 'blue', size = 'md' %}
+
+     {% set alert = cva('alert', {
+        color: {
+            blue: 'bg-blue',
+            red: 'bg-red',
+            green: 'bg-green',
+        },
+        size: {
+            sm: 'text-sm',
+            md: 'text-md',
+            lg: 'text-lg',
+        }
+    }) %}
+
+    <div class="{{ alert.apply({color, size}) ~ ' ' ~ attributes.render('class') | tailwind_merge }}">
```

Comment:
Does `tailwind_merge` apply to `attributes.render('class')` only? Either way, it's a bit confusing imo.

What about rendering the class above?

```diff
- {% set alert = cva('alert', {
+ {% set alert = cva('alert ' ~ attributes.render('class'), {
```

Then, here:

```suggestion
    <div class="{{ alert.apply({color, size})|tailwind_merge }}">
```

Reply:
Yes good idea, it's cleaner! 👍

Reply:
@CMH-Benny I don't think it's changing something. The only difference is now the outside class gonna be in the center of the other class and before the outside class was at the end.

Reply:
Okay, let me try it with an example and how i understand it, I mean in the end this is not a blocker, I just wasn't sure if we should mention the behavior so that devs can do it the way they need it :) 

Let's say we have component:
```twig
{% props theme='primary', size = 'md' %}
{% set button = cva('btn' ~ ' ' ~ attributes.render('class'), {
         theme: {
             primary: 'bg-blue-500',
             secondary: 'bg-gray-400',
             success: 'bg-green-300',
             warning: 'bg-yellow-500',
             danger: 'bg-red-600',
         },
         size: {
             sm: 'p-2',
             md: 'p-4',
             lg: 'p-8',
         }
     },
     compounds: [
         {
             theme: ['danger'],
             size: ['lg'],
             class: 'font-semibold'
         }
     ]
}) %}

<button {{ button.apply({color, size}, 'some other classes')|tailwind_merge }}>...</button>
```

Now using it somewhere:
```twig
<twig:Button theme="danger" size="sm" class="bg-[#3d3d3d]" />
```

Now, background of that button will still be red and not the custom one - Can be intended, but could also be not intuitive, but since attributes appends classes to base classes, the variant class comes later and probably wins in tailwind_merge over the custom color, or am I wrong here?

Reply:
Ah yes, thanks for the explanation. This is a indeed a problem when using tailwind_merge.

Reply:
Would this work?

```twig
{% props theme='primary', size = 'md' %}
{% set button = cva('btn', {
         theme: {
             primary: 'bg-blue-500',
             secondary: 'bg-gray-400',
             success: 'bg-green-300',
             warning: 'bg-yellow-500',
             danger: 'bg-red-600',
         },
         size: {
             sm: 'p-2',
             md: 'p-4',
             lg: 'p-8',
         }
     },
     compounds: [
         {
             theme: ['danger'],
             size: ['lg'],
             class: 'font-semibold'
         }
     ]
}) %}

<button {{ button.apply({theme, size}, 'some other classes', attributes.render('class'))|tailwind_merge }}>...</button>
```

The `apply()` method would change to allow for an unlimited number of "class" arguments. It's a bit verbose is my only complaint (assuming that this also would work).

Reply:
> The apply() method would change to allow for an unlimited number of "class" arguments.

I like this!